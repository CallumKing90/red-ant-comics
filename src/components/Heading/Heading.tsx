import React from "react";

interface Heading {
  variation?: string;
  className?: string;
  children: React.ReactChild;
}

export function Heading({ className, variation, children }: Heading) {
  switch (variation) {
    case "h1":
      return <h1 className={className}>{children}</h1>;
    case "h2":
      return <h2 className={className}>{children}</h2>;
    default:
      return <p className={className}>{children}</p>;
  }
}
