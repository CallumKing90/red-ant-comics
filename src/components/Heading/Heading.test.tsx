import React from "react";
import { Heading } from "./Heading";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Heading", () => {
  it("should render heading with text", () => {
    const exampleText = "Example text";
    const { getByText } = render(<Heading>{exampleText}</Heading>);

    const heading = getByText(exampleText);
    expect(heading).toBeInTheDocument();
    expect(heading).toMatchSnapshot();
  });

  it("should render heading with class name", () => {
    const exampleClass = "exampleclass";
    const { getByText } = render(
      <Heading className={exampleClass}>My heading</Heading>
    );

    const heading = getByText("My heading");
    expect(heading).toHaveClass(exampleClass);
    expect(heading).toMatchSnapshot();
  });

  const headingTypes = ["h1", "h2"];

  for (const headingType of headingTypes) {
    it(`should render ${headingType} variation`, () => {
      const { container } = render(
        <Heading variation={headingType}>My heading</Heading>
      );
      const heading = container.querySelector(headingType);
      expect(heading).toHaveTextContent("My heading");
      expect(heading).toMatchSnapshot();
    });
  }

  it(`should render paragraph when no variation passed`, () => {
    const { container } = render(<Heading>My heading</Heading>);
    const heading = container.querySelector("p");
    expect(heading).toHaveTextContent("My heading");
    expect(heading).toMatchSnapshot();
  });
});
