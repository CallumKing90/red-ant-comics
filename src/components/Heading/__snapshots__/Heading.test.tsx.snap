// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render h1 variation 1`] = `
<h1>
  My heading
</h1>
`;

exports[`Heading should render h2 variation 1`] = `
<h2>
  My heading
</h2>
`;

exports[`Heading should render heading with class name 1`] = `
<p
  class="exampleclass"
>
  My heading
</p>
`;

exports[`Heading should render heading with text 1`] = `
<p>
  Example text
</p>
`;

exports[`Heading should render paragraph when no variation passed 1`] = `
<p>
  My heading
</p>
`;
